activejob/test/cases/argument_serialization_test.rb

```diff
@@ -50,6 +52,24 @@ class ClassArgument; end
     end
   end
 
+  test "dangerously treats BigDecimal arguments as primitives not requiring serialization by default" do
+    assert_deprecated(<<~MSG.chomp) do
+      Primitive serialization of BigDecimal job arguments is deprecated as it may serialize via .to_s using certain queue adapters.
+      Enable config.active_job.use_big_decimal_serializer to use BigDecimalSerializer instead.
+    MSG
+      assert_equal(
+        BigDecimal(5),
+        *ActiveJob::Arguments.deserialize(ActiveJob::Arguments.serialize([BigDecimal(5)])),
+      )
+    end
+  end
+
+  test "safely serializes BigDecimal arguments if configured to use_big_decimal_serializer" do
+    with_big_decimal_serializer do
+      assert_arguments_unchanged BigDecimal(5)
+    end
+  end
+
```

Comment:
In Rails 7.2 these two tests should be deleted, and the `BigDecimal` example should be added back to the list above.

I am unsure of if Rails has a mechanism to enforce this in tests. In Shopify/job-iteration I introduced a [`skip_until_version("2.0")` helper](https://github.com/Shopify/job-iteration/commit/d3fdeff26ed704cfdb4d8232294ba9f376a82e95). A similar `flunk_once_version("7.2")` helper could be added here if desired, but I assume Rails already has a process by which removes deprecated settings in new versions. 🤷